lib/nuget-parser/package-reference-parser.js

```diff
@@ -0,0 +1,55 @@
+'use strict';
+const debug = require('debug')('snyk');
+const _ = require('lodash');
+
+function validateManifest(manifest) {
+  if (!manifest.project) {
+    throw new Error('Project field was not found in project.assets.json');
+  }
+
+  if (!manifest.project.frameworks) {
+    throw new Error('No frameworks were found in project.assets.json');
+  }
+
+  if (_.isEmpty(manifest.project.frameworks)) {
+    throw new Error('0 frameworks were found in project.assets.json');
+  }
+
+  if (!manifest.targets) {
+    throw new Error('No targets were found in project.assets.json');
+  }
+
+  if (_.isEmpty(manifest.targets)) {
+    throw new Error('0 targets were found in project.assets.json');
+  }
+}
+
+function parseProjectAssetsFileContent(fileContent, tree, useProjectNameFromAssetsFile) {
+  debug('Trying to parse package reference manifest');
+
+  let manifest = JSON.parse(fileContent);
+  try {
+    validateManifest(manifest);
+  } catch (error) {
+    debug('Invalid project.assets.json manifest file');
+    throw (error);
+  }
+
+  if (manifest.project.version) {
+    tree.version = manifest.project.version;
+  }
+
+  if (useProjectNameFromAssetsFile) {
+    if (manifest.project.restore.projectName) {
```

Comment:
```suggestion
    if (_.get(manifest.project, 'restore.projectName')) {
```
This will prevent erring if manifest.project.restore is undefined